webapp/src/lib/Ui.ts

```diff
@@ -73,6 +102,121 @@ export interface IDialogConfig {
 export class DialogOKCancelComponent {
     constructor(
         public dialogRef: MatDialogRef<DialogOKCancelComponent>,
-        @Inject(MAT_DIALOG_DATA) public data: IDialogConfig) {
+        @Inject(MAT_DIALOG_DATA) public data: IDialogOKCConfig) {
+    }
+}
+
+export interface IDialogTransactionConfig {
+    title: string;
+    worker: TWorker;
+}
+
+@Component({
+    selector: "app-dialog-transaction",
+    styleUrls: ["./app-dialog-transaction.scss"],
+    templateUrl: "dialog-transaction.html",
+})
+export class DialogTransactionComponent implements OnInit {
+
+    static progress(dtc: DialogTransactionComponent, percentage: number, text: string) {
+        dtc.percentage = percentage >= 0 ? percentage : percentage * -1;
+
+        if (dtc.percentage >= 100) {
+            clearTimeout(dtc.ubTimer);
+            dtc.blockIndex++;
+            dtc.addBlock();
+            setTimeout(() => {
+                dtc.dialogRef.close();
+            }, 1000);
+        }
+
+        if (percentage >= 0) {
+            dtc.text = "";
+            dtc.addTransaction(text);
+        } else {
+            dtc.text = text;
+        }
+    }
+
+    percentage: number;
+    blockIndex: number;
+    blocks: Element[] = [];
+    transaction: Element;
+    error: string;
+    text: string;
+    @ViewChild("main", {static: false}) main?: ElementRef;
+    ubTimer: Timer;
+
+    constructor(
+        public dialogRef: MatDialogRef<DialogTransactionComponent>,
+        public bcv: BcviewerService,
```

Comment:
And some `readonly` are pretty useful also, to avoid modifying fields not meant to.